x-pack/plugins/screenshotting/server/screenshots/wait_for_render.ts

```diff
@@ -22,54 +21,35 @@ export const waitForRenderComplete = async (
     'wait'
   );
 
-  return await browser
-    .evaluate(
-      {
-        fn: (selector, visLoadDelay) => {
-          // wait for visualizations to finish loading
-          const visualizations: NodeListOf<Element> = document.querySelectorAll(selector);
-          const visCount = visualizations.length;
-          const renderedTasks = [];
-
-          function waitForRender(visualization: Element) {
-            return new Promise<void>((resolve) => {
-              visualization.addEventListener('renderComplete', () => resolve());
-            });
-          }
-
-          function waitForRenderDelay() {
-            return new Promise((resolve) => {
-              setTimeout(resolve, visLoadDelay);
-            });
+  await browser.evaluate(
+    {
+      fn: async (selector) => {
+        const visualizations: NodeListOf<Element> = document.querySelectorAll(selector);
+        const visCount = visualizations.length;
+        const renderedTasks = [];
+
+        function waitForRender(visualization: Element) {
+          return new Promise<void>((resolve) => {
+            visualization.addEventListener('renderComplete', () => resolve());
+          });
+        }
+
+        for (let i = 0; i < visCount; i++) {
+          const visualization = visualizations[i];
+          const isRendered = visualization.getAttribute('data-render-complete');
+
+          if (isRendered === 'false') {
+            renderedTasks.push(waitForRender(visualization));
           }
+        }
 
-          for (let i = 0; i < visCount; i++) {
-            const visualization = visualizations[i];
-            const isRendered = visualization.getAttribute('data-render-complete');
-
-            if (isRendered === 'disabled') {
-              renderedTasks.push(waitForRenderDelay());
-            } else if (isRendered === 'false') {
-              renderedTasks.push(waitForRender(visualization));
-            }
-          }
-
-          // The renderComplete fires before the visualizations are in the DOM, so
-          // we wait for the event loop to flush before telling reporting to continue. This
-          // seems to correct a timing issue that was causing reporting to occasionally
-          // capture the first visualization before it was actually in the DOM.
-          // Note: 100 proved too short, see https://github.com/elastic/kibana/issues/22581,
-          // bumping to 250.
-          const hackyWaitForVisualizations = () => new Promise((r) => setTimeout(r, 250));
```

Comment:
`hackyWaitForVisualizations` was able to be removed without creating flakiness in the tests.